File: 12A-Labs/lab8/GCD.c
/*
 * GCD,c
 * Navjot Heer
 * nsheer
 * lab8
 * Prints the greatest common denominator inserted
 * by the user.        
 *        
 */        

#include<stdio.h>
#include<stdlib.h>

int main(){
    int b, n, x, y, div, remainder;
    int large = 0;
    int small = 0;
    
    printf("Enter a positive integer: ");
    n = scanf("%d", &x);
    while ( 1 ){
        n = scanf(" %d", &x);
        while( n!=1){
            scanf("%s", &b);
            printf("Please enter a positive integer: ");
            n = scanf("%d", &x);
        }
        if( x > 0 ) break;
            printf("Please enter a positive integer: ");
    }
        
   printf("Enter another positive integer: ");
   n = scanf("%d", &x);
    while ( 1 ){
        n = scanf(" %d", &y);
        while( n!=1){
            scanf("%s", &b);
            printf("Please enter a positive integer: ");
            n = scanf("%d", &y);
        }
        if( y > 0 ) break;
            printf("Please enter a positive integer: ");
    }
        
        if ( x > y ){
             large = x;
             small = y;
        }
        
        else if ( x < y) {
        small = x;
        large = y;

        }

   div = large/small;
   remainder = large%small;

    while( large != 0 && small != 0){
       remainder = small;
       small = large%small;
       large = remainder;
    }

  printf("The GCD of %d and %d is %d.\n", x, y,  remainder);

  return 0;
}

File: 12A-Labs/lab4/GCD.java
//pa3
//Navjot Heer
//nsheer
//GCD.java
//Prints the Greatest Common Denominator based on two
//postive inputs by the user

import java.util.Scanner;
class GCD {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int x, y, div, remainder;
        int large = 0;
        int small = 0;

        System.out.print("Enter a positive integer: ");
        while(true){
            if(sc.hasNextInt()){
                x = sc.nextInt();
                if(x > 0){
                    break;
                }
            }

            System.out.print("Please enter a positive integer: ");
            sc.next();
        }

        System.out.print("Enter another positive integer: ");
        while(true){

            if(sc.hasNextInt()){
                y = sc.nextInt();
                if(y > 0){
                    break;
                }
            }

            System.out.print("Please enter a positive integer: ");
            sc.next();
        }

        if (x > y) {
            large = x;
            small = y;
        }
        else if (x < y) {
            small = x;
            large = y;
        }
        div = large / small;
        remainder = large % small;
        while (large != 0 && small != 0) {
            remainder = small;
            small = large % small;
            large = remainder;
        }
        System.out.println("The GCD of " + x + " and " + y + " is " + remainder);
    }
}


